pyecoregen/templates/module_utilities.tpl

```diff
@@ -77,8 +88,12 @@ class Derived{{ d.name | capitalize }}(EDerivedCollection):
 
 {%- macro generate_feature_init(feature) %}
     {%- if feature.upperBound == 1 %}
+        {%- if not textX %}
         if {{ feature.name }} is not None:
             self.{{ feature.name }} = {{ feature.name }}
+        {%- else %}
```

Comment:
This `if` statement removal breaks the default value capabilities of `EStructuralFeatures`, is there a reason why the affectation is mandatory? Perhaps textX directly works with instance `__dict__` sometimes?

Reply:
My custom scope providers break, because instead of `None` I sometimes have a `property` object.

Consider this grammar:
```
TypeReference:
	predefined=BasicType | derived=[Type];

BasicType:
	Boolean = /Boolean(?=\W)/ |
	String = /String(?=\W)/ |
	Integer = /Integer(?=\W)/ ;

Type: StructType | TypeDefinition;
```

A resolved TypeReference with a `derived` reference looks like this:
```
TypeReference: {
  predefined: <property object>
  derived: <TypeDefinition object>
}
```

Is this the intended behavior?

Reply:
I cannot, but here you can see a small working example:
https://github.com/fgro93/pyecore-Provider

```
class ProblemRuleValProvider():
    def __call__(self, obj, attr, obj_ref):
        reference = obj.parent.reference
        if reference.predefined:
            # imagine meaningful code here
            obj.attribute = 42
        print(hasattr(obj, 'attribute')) # prints True
        return reference
```

Reply:
I see. Sounds like a good solution to me. :)

I agree, code changes by  `--textX` should be kept to a minimum.